File: P5Java/Napakalaki/src/napakalaki/Monster.java
package napakalaki;

public class Monster implements Card{
    private String name;
    private int combatLevel;
    private BadConsequence bc;
    private Prize price;
    private String icon;
    private int levelChangeAgainstCultistPlayer;

    
    public Monster(String name, int combatLevel, BadConsequence bc, Prize price, String icon) {
        this.name = name;
        this.combatLevel = combatLevel;
        this.bc = bc;
        this.price = price;
        this.icon = icon;
    }
    //TODO he creado otro constructor, porque si no peta en los no cultists
    public Monster(String name, int combatLevel, BadConsequence bc, Prize price, String icon, int levelChangeACultistPlayer) {
        this.name = name;
        this.combatLevel = combatLevel;
        this.bc = bc;
        this.price = price;
        this.icon = icon;
        this.levelChangeAgainstCultistPlayer = levelChangeACultistPlayer;
    }

    public int getCombatLevel() {
        return combatLevel;
    }

    public String getName() {
        return name;
    }

    public BadConsequence getBc() {
        return bc;
    }

    public Prize getPrice() {
        return price;
    }

    public String getIcon() {
        return icon;
    }
    
    public int getLevelsGained(){
        return price.getLevel();
    }
    
    public int getTreasuresGained(){
        return price.getTreasures();
    }
    
    public boolean kills(){
        return bc.myBadConsequenceisDeath();
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setCombatLevel(int combatLevel) {
        this.combatLevel = combatLevel;
    }

    public void setBc(BadConsequence bc) {
        this.bc = bc;
    }

    public void setPrice(Prize price) {
        this.price = price;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }
    
    
    public String toString() {
        return "Name = "+getName()+" combatLevel = "+Integer.toString(combatLevel) + "Bc = "+getBc().toString();
    }

    @Override
    public int getBasicValue() {
        return this.getCombatLevel();
    }

    @Override
    public int getSpecialValue() {
        return this.getCombatLevel() + this.levelChangeAgainstCultistPlayer;
    }
    
    
}

File: P5Ruby/NapakalakiRuby/lib/monster.rb
#encoding: utf-8

require_relative "card"
class Monster < Card
  
  attr_accessor :name, :combat_level, :prize, :bc, :level_change_against_cultist_player
  
  def initialize(name,combat_level, prize, bc,lcacp)
    @name = name
    @combat_level = combat_level
    @prize = prize
    @bc = bc
    @level_change_against_cultist_player = lcacp
  end
  
  def get_levels_gained
    return @prize.level
  end
  
  def get_treasures_gained
    return @prize.treasures
  end
  
  def kill
    return @bc.my_bad_consequence_is_death
  end
  
  def get_basic_value
    return @combat_level
  end
  
  def get_special_value
    return @combat_level + @level_change_against_cultist_player
  end
  public
  def to_s
    "Monstruo: #{@name}, Mal rollo: #{@bc}"
    
  end
end
